### PR TITLE
[UPDATED]Makes Foam Force mags sprites unique, adds a QoL change to foam force

### PR DESCRIPTION
**Pull request by Tacolizard**
:cl: NT Manufacturing Division
tweak: Foam Force magazine sprites have been updated.
add: Foam dart caps can be unscrewed by hand.
/:cl:

~~the ammo boxes now have their ammo type written on them~~ **The ammo box sprites have been reverted**, and the foam force pistol uses a nerf-themed mag

### CODE REVIEW

**Comment by optimumtact** [comment]
macdonalds lawyers gonna be giving us a call bout those golden arches :joy:

**Comment by QualityVan** [comment]
I'm not huge on the ammo box sprites.

**Comment by Lumipharon** [comment]
Yeah, to be honest those box sprites are pretty ugly - besides, why do they need their calibre printed on them? You can check just by mouse hover/examine, they're already colour coded, and if nothing else, you are unlikely to even need multiple types of ammo at any one time.

**Comment by Tacolizard** [comment]
Just to make it clear, those are the same box sprites we've had for a while. I only added the lettering on top.

**Comment by RandomMarine** [comment]
I hate to rain on your parade, but I was working on these IN SECRET.
boolits

**Comment by WJohn** [comment]
Yeaaah these are not wonderful, honestly. I think the color coded ones we have now are fairly recognizable. At least shrink down the text.

**Comment by Tacolizard** [comment]
**It's only failing because some other pr broke everything**

**Comment by Lumipharon** [comment]
They don't need writing on them just like different medkits don't need their type written on them - being colour coded and literally having the type in their name is enough, without bloating up the sprite.

**Comment by Tacolizard** [comment]
@Lumipharon my latest commit removed the writing on them.

File: code/modules/projectiles/boxes_magazines/external_mag.dm [hunk 1]
```diff
@@ -346,13 +346,20 @@
...
+	icon_state = "9x19p-8-toy"
 	max_ammo = 8
 	multiple_sprites = 2
 
 /obj/item/ammo_box/magazine/toy/pistol/riot
 	ammo_type = /obj/item/ammo_casing/caseless/foam_dart/riot
 
+/obj/item/ammo_box/magazine/toy/pistol/update_icon()
+	..()
+	if(ammo_count())
```
Comment (lzimann) [inline code/modules/projectiles/boxes_magazines/external_mag.dm]: You can one line this with
`icon_state = "9x19-[ammo_count() ? "8" : "0"]-toy"`

File: code/modules/projectiles/ammunition/caseless.dm [hunk 2]
```diff
@@ -102,6 +102,12 @@
 
 /obj/item/ammo_casing/caseless/foam_dart/attack_self(mob/living/user)
 	var/obj/item/projectile/bullet/reusable/foam_dart/FD = BB
+	if (!modified)
+		modified = 1
```
Comment (lzimann) [inline code/modules/projectiles/ammunition/caseless.dm]: `TRUE` for both

**Comment by lzimann** [comment]
Also please update with the current icons

**Comment by Tacolizard** [comment]
@lzimann your suggestion with the icon state is causing a compile error. I probably implemented it wrong though.

**Comment by lzimann** [comment]
It has to be done on update_icon().

**Pull request closed without merging**